app/models/katello/concerns/smart_proxy_extensions.rb

```diff
@@ -139,14 +139,18 @@ def pulp3_enabled?
       end
 
       def pulp3_support?(repository)
-        type = Katello::RepositoryTypeManager.repository_types[repository.content_type]
-        type.pulp3_plugin.present? && pulp3_enabled?
+        pulp3_repository_type_support?(repository.content_type)
       end
 
-      def content_pulp3_support?(content_type)
-        content_type = content_type.is_a?(String) ? content_type : content_type.model_class::CONTENT_TYPE
-        type = Katello::RepositoryTypeManager.find_repository_type content_type
-        type.pulp3_plugin && SmartProxy.pulp_master!.capabilities('Pulp3').try(:include?, type.pulp3_plugin)
+      def pulp3_repository_type_support?(repository_type)
+        repository_type = Katello::RepositoryTypeManager.repository_types[repository_type] if repository_type.is_a?(String)
```

Comment:
Should we be checking for `nil` here? When I tested with an unknown repo type, it errored out on the next line with `NoMethodError (undefined method `pulp3_plugin' for nil:NilClass)`